web/src/components/TaskHeader.test.tsx

```diff
@@ -192,6 +193,456 @@ describe("<TaskHeader />", () => {
+      selectCompleted();
+
+      expect(screen.getByText(Config.profileDefaults.STARTING.ownership.header)).toBeInTheDocument();
+      expect(
+        screen.getByText(markdownToText(Config.profileDefaults.STARTING.existingEmployees.header))
+      ).toBeInTheDocument();
+    });
+
+    it("shows ownership and employee count for gp", () => {
+      const taxTask = generateTask({ id: taxTaskId, stepNumber: 1 });
+
+      const userData = generateUserData({
+        profileData: generateProfileData({
+          businessName: "",
+          legalStructureId: "general-partnership",
+        }),
+        taskProgress: {
+          [taxTaskId]: "NOT_STARTED",
+        },
+        preferences: generatePreferences({ roadmapOpenSections: ["PLAN"] }),
+      });
+
+      useMockRoadmap({
+        steps: [generateStep({ stepNumber: 1, section: "PLAN" })],
+        tasks: [taxTask],
+      });
+
+      renderTaskHeader(taxTask, userData);
+      selectCompleted();
+
+      expect(screen.getByText(Config.profileDefaults.STARTING.ownership.header)).toBeInTheDocument();
+      expect(
+        screen.getByText(markdownToText(Config.profileDefaults.STARTING.existingEmployees.header))
+      ).toBeInTheDocument();
+    });
+
+    it("shows the warning modal if the user tries to change status from completed", async () => {
+      const taxTask = generateTask({ id: taxTaskId, stepNumber: 1 });
+
+      const userData = generateUserData({
+        profileData: generateProfileData({
+          businessName: "test-business",
+          legalStructureId: "general-partnership",
+        }),
+        taskProgress: {
+          [taxTaskId]: "COMPLETED",
+        },
+        preferences: generatePreferences({ roadmapOpenSections: ["PLAN"] }),
+      });
+
+      useMockRoadmap({
+        steps: [generateStep({ stepNumber: 1, section: "PLAN" })],
+        tasks: [taxTask],
+      });
+
+      renderTaskHeader(taxTask, userData);
+      fireEvent.click(screen.getByTestId("taskProgress"));
+      fireEvent.click(screen.getByTestId("IN_PROGRESS"));
+
+      expect(screen.getByText(Config.taxRegistrationModal.areYouSureTaxBody)).toBeInTheDocument();
+    });
+
+    it("updates the task progress if the user continues in the warning modal", async () => {
+      const taxTask = generateTask({ id: taxTaskId, stepNumber: 1 });
+
+      const userData = generateUserData({
+        profileData: generateProfileData({
+          businessName: "test-business",
+          legalStructureId: "general-partnership",
+        }),
+        taskProgress: {
+          [taxTaskId]: "COMPLETED",
+        },
+        preferences: generatePreferences({ roadmapOpenSections: ["PLAN"] }),
+      });
+
+      useMockRoadmap({
+        steps: [generateStep({ stepNumber: 1, section: "PLAN" })],
+        tasks: [taxTask],
+      });
+
+      renderTaskHeader(taxTask, userData);
+      fireEvent.click(screen.getByTestId("taskProgress"));
+      fireEvent.click(screen.getByTestId("IN_PROGRESS"));
```

Comment:
the other tests in this file use getAllByText instead to select these. Is it intentional to switch to a testid model? What do you prefer, or think the tradeoffs are?